### ISSUE TITLE
What will be done with FM's outputs?

### ISSUE THREAD

**Issue by ghost** [DJEddyking/LF2#2]
Hi @DJEddyking

I have be interested in your project, thus I have dig into your code. Then I've found that the fuse_outputs maybe be useless. 
Code in [trainers.py](https://github.com/DJEddyking/LF2/blob/833dc38546c7fbb999ff0c4843f395a70b58dc8c/reid/trainers.py#L127)
` feat_list, prob, prob_ema, fuse_outputs  = self.encoder(inputs_1)`
 `feat_list_2, prob_2, prob_2_ema, fuse_outputs_2 = self.encoder(inputs_2) `
            `feat_list_3, prob_3, prob_3_ema, fuse_outputs_3 = self.encoder(inputs_3)`

Is my speculation incorrect? And What does FM do in your code?
Thanks!

**Comment by DJEddyking** [comment]
Thanks for following our work. FM is used to generate more reliable pseudo labels for optimizing the student network. During the training, for the local feature of the student network, we use the pseudo labels generated by by corresponding fusion feature to calculate the softmax triplet loss. More details in the subsection _C.Target-domain fine-tuning_ of our paper: [Learning Feature Fusion for Unsupervised Domain Adaptive Person Re-identification](https://arxiv.org/abs/2205.09495)

**Comment by ewigspace1910** [comment]
> Thanks for following our work. FM is used to generate more reliable pseudo labels for optimizing the student network. During the training, for the local feature of the student network, we use the pseudo labels generated by by corresponding fusion feature to calculate the softmax triplet loss. More details in the subsection _C.Target-domain fine-tuning_ of our paper: [Learning Feature Fusion for Unsupervised Domain Adaptive Person Re-identification](https://arxiv.org/abs/2205.09495)

Yep, I have read this marvelous paper, I have got ideal about FM in your paper and found that you use FM  in the function `extract_cnn_feature_new`

But I am still ambiguous for understanding your code in [trainers.py](https://github.com/DJEddyking/LF2/blob/833dc38546c7fbb999ff0c4843f395a70b58dc8c/reid/trainers.py#L127). As the above mention, `fuse_ouputs, fuse_ouputs_2 and fuse_outputs_3` in `SingelmeanTrainer` hmm maybe never be used. Is it redundant?


Another question, why do you take two output from image and it's flip? Is there new technique? ( I'm novice :<)
`       outputs = extract_cnn_feature_new(model, imgs)`
        `outputs_ = extract_cnn_feature_new(model,  fliplr(imgs))`

**Comment by DJEddyking** [comment]
R1. Thank you for your reminder. `fuse_ouputs, fuse_ouputs_2 and fuse_outputs_3`  is not redundant. We have conducted relevant experiments and the results are better than current ones. The latest code will be released soon in the second version.
R2. It is a trick for evaluation. It can increase the discriminability of the model at the feature level to a certain extent.
